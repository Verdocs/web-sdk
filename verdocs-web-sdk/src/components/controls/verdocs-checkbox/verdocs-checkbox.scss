@import '../../../theme.scss';

// Your periodic reminder that it's 2022 and you still can't style HTML checkboxes
verdocs-checkbox {
  font-family: $verdocs-primary-font;
  line-height: 18px;

  label > input[type='checkbox'] {
    display: none;
  }

  label > input[type='checkbox'] + *::before {
    content: '';
    width: 1rem;
    height: 1rem;
    flex-shrink: 0;
    line-height: 18px;
    border-radius: 2px;
    border-style: solid;
    margin-right: 0.5rem;
    border-width: 0.1rem;
    display: inline-block;
    vertical-align: bottom;
    border-color: $verdocs-grey-2;
  }

  label > input[type='checkbox']:checked + * {
    color: $verdocs-green;
  }

  label > input[type='checkbox']:checked + *::before {
    content: 'âœ“';
    color: white;
    text-align: center;
    background: $verdocs-green;
    border-color: $verdocs-green;
  }

  &.dark {
    label > input[type='checkbox'] + *::before {
      border-color: $verdocs-grey-4;
    }

    label > input[type='checkbox']:checked + *::before {
      background: $verdocs-green;
      border-color: $verdocs-green;
    }
  }
}
