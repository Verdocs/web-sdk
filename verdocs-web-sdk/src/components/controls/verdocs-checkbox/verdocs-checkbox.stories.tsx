import {html} from 'lit-html';
import {Meta} from '@storybook/web-components';

export default {
  title: 'Controls/Checkbox',
  component: 'verdocs-checkbox',
  args: {
    checked: false,
    name: 'test',
    value: '1',
    disabled: false,
  },
  argTypes: {},
} as Meta;

export const Checkbox = ({checked, name, value, disabled}) => html` <verdocs-checkbox .checked=${checked} .name=${name} .value=${value} .disabled=${disabled} /> `;
