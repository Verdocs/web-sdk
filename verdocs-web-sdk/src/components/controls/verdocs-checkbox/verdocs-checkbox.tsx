import {Component, h, Host, Prop, Event, EventEmitter} from '@stencil/core';

/**
 * Displays a radio button. Note that this is different from the `verdocs-field-radio-button` component, which is designed
 * to be used in signing experiences and contains settings that connect to template fields. This is just a simple radio
 * button for UI displays e.g. dialogs and .
 */
@Component({
  tag: 'verdocs-checkbox',
  styleUrl: 'verdocs-checkbox.scss',
  shadow: false,
})
export class VerdocsCheckbox {
  /**
   * Whether the radio button is currently selected.
   */
  @Prop() checked: boolean = false;

  /**
   * HTML form field name for the input.
   */
  @Prop() name: string = '';

  /**
   * Value to track with the input.
   */
  @Prop() value: string = '';

  /**
   * If set, the button will still be displayed but not selectable.
   */
  @Prop() disabled?: boolean = false;

  /**
   * Event fired when the input field value changes. Note that this will only be fired on blur, tab-out, ENTER key press, etc.
   * It is generally the best event to subscribe to than `input` for most cases EXCEPT autocomplete fields that need to see every
   * keypress.
   */
  @Event({composed: true}) selected: EventEmitter<{value: string}>;

  handleChange(e: any) {
    console.log('changed', e.target.checked);
    this.selected.emit({value: this.value});
  }

  render() {
    return (
      <Host class={{disabled: this.disabled}}>
        <input
          type="checkbox"
          value={this.value}
          name={this.name}
          id={`verdocs-checkbox-${this.name}-${this.value}`}
          checked={this.checked}
          disabled={this.disabled}
          onChange={e => this.handleChange(e)}
        />
      </Host>
    );
  }
}
