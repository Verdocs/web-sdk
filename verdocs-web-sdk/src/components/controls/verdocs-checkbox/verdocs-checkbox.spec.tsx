import { newSpecPage } from '@stencil/core/testing';
import { VerdocsCheckbox } from './verdocs-checkbox';

describe('verdocs-checkbox', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsCheckbox],
      html: `<verdocs-checkbox></verdocs-checkbox>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with checked, theme, size, and label', async () => {
    const page = await newSpecPage({
      components: [VerdocsCheckbox],
      html: `<verdocs-checkbox checked="true" theme="dark" size="small" label="Test"></verdocs-checkbox>`,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root.querySelector('input[type="checkbox"]')?.checked).toBe(true);
  });

  it('handles input event and toggles checked state', async () => {
    const page = await newSpecPage({
      components: [VerdocsCheckbox],
      html: `<verdocs-checkbox></verdocs-checkbox>`,
    });
    const input = page.root.querySelector('input[type="checkbox"]');
    expect(input).not.toBeNull();
    input.checked = true;
    input.dispatchEvent(new Event('input'));
    // Stencil's @Prop() does not auto-update, but we can check the event fired
    // For a real test, you would listen for a custom event if implemented
    expect(input.checked).toBe(true);
  });
});
