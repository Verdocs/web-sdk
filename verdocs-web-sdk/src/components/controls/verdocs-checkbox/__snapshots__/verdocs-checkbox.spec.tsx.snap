// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-checkbox renders with checked, theme, size, and label 1`] = `
<verdocs-checkbox checked="true" class="dark small" label="Test" size="small" theme="dark">
  <label>
    <input checked="" type="checkbox" value="">
    <span>
      Test
    </span>
  </label>
</verdocs-checkbox>
`;

exports[`verdocs-checkbox renders with default props 1`] = `
<verdocs-checkbox class="light normal">
  <label>
    <input type="checkbox" value="">
    <span></span>
  </label>
</verdocs-checkbox>
`;
